### ISSUE TITLE
Wrapper Cookbook and Resources Questions

### ISSUE THREAD

**Issue by jimdaga** [sous-chefs/postgresql#527]
### Cookbook version
postgresql (6.1.4)

### Chef-client version
Chef: 13.6.4

### Platform Details
ubuntu-trusty-64 (Vagrant & GCP)

### Scenario:
Wrapper Cookbook with Chef-Solo (or chef-client -z), not finding resources from sous-chefs/postgresql cookbook.

### Steps to Reproduce:

Create a test wrapper cookbook.
```
$ cd /var/chef/cookbooks/
$ chef generate cookbook test_postgresql
```

Update the default recipe to configure any of the postgresql resources.
```
$ vim test_postgresql/recipes/default.rb

  #
  # Cookbook:: test_postgresql
  # Recipe:: default
  #
  # Copyright:: 2018, The Authors, All Rights Reserved.

  postgresql_client_install 'Postgresql Client' do
    setup_repo false
    version '9.5'
  end
```

Update the wrapper cookbook metadata to use the postgresql cookbook.
```
$ grep depends test_postgresql/metadata.rb
  depends 'postgresql'
```

Update Berks to use postgresql cookbook.
```
$ grep cookbook test_postgresql/Berksfile
  cookbook 'postgresql', '~> 6.1.4'
```

Vendor cookbooks into same cookbook directory.
```
$ berks vendor -b test_postgresql/Berksfile .
```

Converge using wrapper cookbook.
```
$ sudo chef-solo -o 'test_postgresql::default'
[2018-05-01T18:23:54+00:00] WARN: *****************************************
[2018-05-01T18:23:54+00:00] WARN: Did not find config file: /etc/chef/client.rb, using command line options.
[2018-05-01T18:23:54+00:00] WARN: *****************************************
[2018-05-01T18:23:54+00:00] INFO: Started chef-zero at chefzero://localhost:1 with repository at /var/chef
  One version per cookbook

[2018-05-01T18:23:54+00:00] INFO: Forking chef instance to converge...
Starting Chef Client, version 13.6.4
[2018-05-01T18:23:54+00:00] INFO: *** Chef 13.6.4 ***
[2018-05-01T18:23:54+00:00] INFO: Platform: x86_64-linux
[2018-05-01T18:23:54+00:00] INFO: Chef-client pid: 21394
[2018-05-01T18:23:54+00:00] INFO: The plugin path /etc/chef/ohai/plugins does not exist. Skipping...
[2018-05-01T18:23:56+00:00] WARN: Run List override has been provided.
[2018-05-01T18:23:56+00:00] WARN: Original Run List: []
[2018-05-01T18:23:56+00:00] WARN: Overridden Run List: [recipe[test_postgresql::default]]
[2018-05-01T18:23:56+00:00] INFO: Run List is [recipe[test_postgresql::default]]
[2018-05-01T18:23:56+00:00] INFO: Run List expands to [test_postgresql::default]
[2018-05-01T18:23:56+00:00] INFO: Starting Chef Run for vagrant-ubuntu-trusty-64
[2018-05-01T18:23:56+00:00] INFO: Running start handlers
[2018-05-01T18:23:56+00:00] INFO: Start handlers complete.
resolving cookbooks for run list: ["test_postgresql::default"]
[2018-05-01T18:23:56+00:00] INFO: Loading cookbooks [test_postgresql@0.1.0, postgresql@6.1.4, compat_resource@12.19.1, build-essential@8.1.1, openssl@8.1.2, seven_zip@2.0.2, mingw@2.0.2, windows@4.2.2]
[2018-05-01T18:23:56+00:00] INFO: Skipping removal of obsoleted cookbooks from the cache
Synchronizing Cookbooks:
  - test_postgresql (0.1.0)
  - openssl (8.1.2)
  - postgresql (6.1.4)
  - seven_zip (2.0.2)
  - compat_resource (12.19.1)
  - build-essential (8.1.1)
  - windows (4.2.2)
  - mingw (2.0.2)
Installing Cookbook Gems:
Compiling Cookbooks...

================================================================================
Recipe Compile Error in /etc/chef/local-mode-cache/cache/cookbooks/test_postgresql/recipes/default.rb
================================================================================

NoMethodError
-------------
undefined method `postgresql_client_install' for cookbook: test_postgresql, recipe: default :Chef::Recipe

Cookbook Trace:
---------------
  /etc/chef/local-mode-cache/cache/cookbooks/test_postgresql/recipes/default.rb:7:in `from_file'

Relevant File Content:
----------------------
/etc/chef/local-mode-cache/cache/cookbooks/test_postgresql/recipes/default.rb:

  1:  #
  2:  # Cookbook:: test_postgresql
  3:  # Recipe:: default
  4:  #
  5:  # Copyright:: 2018, The Authors, All Rights Reserved.
  6:
  7>> postgresql_client_install 'Postgresql Client' do
  8:    setup_repo false
  9:    version '9.5'
 10:  end
 11:

System Info:
------------
chef_version=13.6.4
platform=ubuntu
platform_version=14.04
ruby=ruby 2.4.2p198 (2017-09-14 revision 59899) [x86_64-linux]
program_name=chef-solo worker: ppid=21389;start=18:23:54;
executable=/opt/chefdk/bin/chef-solo


Running handlers:
[2018-05-01T18:23:56+00:00] ERROR: Running exception handlers
Running handlers complete
[2018-05-01T18:23:56+00:00] ERROR: Exception handlers complete
Chef Client failed. 0 resources updated in 01 seconds
[2018-05-01T18:23:56+00:00] FATAL: Stacktrace dumped to /etc/chef/local-mode-cache/cache/chef-stacktrace.out
[2018-05-01T18:23:56+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
[2018-05-01T18:23:56+00:00] ERROR: undefined method `postgresql_client_install' for cookbook: test_postgresql, recipe: default :Chef::Recipe
[2018-05-01T18:23:56+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```

### Expected Result:
For this test, I had expected the `test_postgresql` cookbook to call the `postgresql` cookbook and run the `resources/client_install.rb` resource.

### Actual Result:
Get a `undefined method postgresql_client_install` error.

### Side Note:
I can get this to work using a chef role and using override_attributes attributes but would like to have some items in my wrapper cookbook that I don't have to define in each role.

**Comment by notapatch** [comment]
Not the developer but the readme on the master branch is for the work in progress resource library version of the cookbook 7.xx. 

Until 7.x.x refer to the readme on the 6.x.x branch and fix the version to 6.1.1 in the metadata of your wrapper cookbook. The versions after 6.1.1 seems to be changed with the new branch, 7.x.x, in mind  (you seem to be using 6.1.4) . - [see warnings](https://github.com/sous-chefs/postgresql/commit/847f166a5f143e7c440fda7474fbe1e7edef67b0).

metadata.rb
`depends          'postgresql', '= 6.1.1'`

**Comment by jimdaga** [comment]
@notapatch thanks for helping clarify that, I had misunderstood that it was still in progress. I'll stick with 6.1.1 with roles for now. Looking forward to the shift to resources, thank you sous-chefs team!
